BraintreeDropIn/BTDropInController.m

```diff
@@ -269,12 +281,56 @@ - (void)cardTokenizationCompleted:(BTPaymentMethodNonce *)tokenizedCard error:(N
                 result.cancelled = YES;
             }
             [sender dismissViewControllerAnimated:YES completion:^{
-                self.handler(self, result, error);
+                if ([self.configuration.json[@"threeDSecureEnabled"] isTrue] && self.dropInRequest.threeDSecureRequest) {
+                    [self threeDSecureVerification:tokenizedCard];
+                } else {
+                    self.handler(self, result, error);
+                }
             }];
         }
     });
 }
 
+- (void)threeDSecureVerification:(BTPaymentMethodNonce *)tokenizedCard {
+    BTPaymentFlowDriver *paymentFlowDriver = [[BTPaymentFlowDriver alloc] initWithAPIClient:self.apiClient];
+    paymentFlowDriver.viewControllerPresentingDelegate = self;
+
+    BTThreeDSecureRequest *request = self.dropInRequest.threeDSecureRequest;
+    if (!request) {
+        request = [[BTThreeDSecureRequest alloc] init];
+    }
+    request.nonce = tokenizedCard.nonce;
+
+    if (!request.amount && self.dropInRequest.amount) {
+        request.amount = [[NSDecimalNumber alloc] initWithString:self.dropInRequest.amount];
+    }
```

Comment:
We're handling a nil `BTThreeDSequreRequest` up on line 84. Here we're handling that case and the nil amount case. Can these checks be consolidated?